OpenRA.Mods.Common/Traits/DockOnRepairCursor.cs

```diff
@@ -0,0 +1,49 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Send actor to dock on repair cursor.")]
+	public class DockOnRepairCursorInfo : TraitInfo, Requires<IDockClientManagerInfo>
+	{
+		[Desc("Docking type")]
+		public readonly BitSet<DockType> Type = new("Repair");
+
+		public override object Create(ActorInitializer init) { return new DockOnRepairCursor(init.Self, this); }
+	}
+
+	public class DockOnRepairCursor
```

Comment:
Why do we have an extra trait for this? This should be part of the dock client?

Reply:
Why should it? I can imagine plenty scenarios where you would want this trait but not Repairable. Let's say in a case where healing is granted via condition

Reply:
It's also not the job of Repairable to give orders, but of DockClientManager

Reply:
Why would healing granted by a condition be covered by this trait? The naming clearly suggests it is about docking. It does not make sense to separate this trait.

Reply:
> Why would healing granted by a condition be covered by this trait?

the actor that provides the healing aura could have a fake dock. It's one way to make repair cursor to work with an alternative healing system